Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -122,34 +125,82 @@ class BackForwardListViewController: UIViewController, UITableViewDataSource, UI
         guard let bvc = self.bvc else {
             return
         }
-        tableView.snp_updateConstraints { make in
-            make.bottom.equalTo(self.view).offset(bvc.shouldShowFooterForTraitCollection(newCollection) ? -UIConstants.ToolbarHeight : 0)
+        if bvc.shouldShowFooterForTraitCollection(newCollection) != snappedToBottom {
+            self.dismissing = true
+            tableView.snp_updateConstraints { make in
+                if snappedToBottom {
+                    make.bottom.equalTo(self.view).offset(0)
+                } else {
+                    make.top.equalTo(self.view).offset(0)
+                }
+                make.height.equalTo(0)
+            }
+            snappedToBottom = !snappedToBottom
         }
     }
     
     override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
         super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
-        tableView.snp_updateConstraints { make in
-            make.height.equalTo(min(CGFloat(BackForwardViewUX.RowHeight*listData.count), size.height/2))
+        let correctHeight = {
+            self.tableView.snp_updateConstraints { make in
+                make.height.equalTo(min(CGFloat(BackForwardViewUX.RowHeight * self.listData.count), size.height / 2))
+            }
+        }
+        if !self.dismissing {
+            correctHeight()
+        } else {
+            coordinator.animateAlongsideTransition(nil) { _ in
+                self.remakeVerticalConstraints()
+                correctHeight()
+                self.dismissing = false
+            }
         }
     }
     
-    override func viewDidLoad() {
-        super.viewDidLoad()
+    func remakeVerticalConstraints() {
         guard let bvc = self.bvc else {
             return
         }
+        for constraint in self.verticalConstraints {
+            constraint.deactivate()
+        }
+        self.verticalConstraints = []
+        tableView.snp_makeConstraints { make in
```

Comment:
> But doesn't snp_remakeConstraints remove all existing constraints? I don't want to remove the horizontal constraints, and I'd have to re-add them if I did it this way, wouldn't I?

Ah true. Makes sense. Removing/re-adding the horizontal constraints wouldn't add any additional overhead though. I think I'm just used to always remaking the constraints when something changes. This works too though!